### ISSUE TITLE
'Error' Classes still seem to persist when loading SWEET

### ISSUE THREAD

**Issue by lewismc** [ESIPFed/sweet#84]
Even after I thought we had [removed the Error Classes from SWEET](https://github.com/ESIPFed/sweet/pull/83), it seems that there are still issues...
Does anyone know how these Classes are being loaded and associated with the Ontology suite? I am unable to find their presence if I grep for them locally or via COR or via Github so I am a bit confused.
```
2018-03-03 17:49:03,664 ERROR (OWLRDFConsumer:1808) Entity not properly recognized, missing triples in input? http://org.semanticweb.owlapi/error#Error1 for type Class
2018-03-03 17:49:03,669 ERROR (OWLRDFConsumer:1808) Entity not properly recognized, missing triples in input? http://org.semanticweb.owlapi/error#Error2 for type Class
2018-03-03 17:49:03,673 ERROR (OWLRDFConsumer:1808) Entity not properly recognized, missing triples in input? http://org.semanticweb.owlapi/error#Error3 for type Class
2018-03-03 17:49:03,674 ERROR (OWLRDFConsumer:1808) Entity not properly recognized, missing triples in input? http://org.semanticweb.owlapi/error#Error4 for type Class
2018-03-03 17:49:03,675 ERROR (OWLRDFConsumer:1808) Entity not properly recognized, missing triples in input? http://org.semanticweb.owlapi/error#Error5 for type Class
2018-03-03 17:49:23,431 ERROR (OWLRDFConsumer:1808) Entity not properly recognized, missing triples in input? http://org.semanticweb.owlapi/error#Error6 for type Class
2018-03-03 17:49:23,433 ERROR (OWLRDFConsumer:1808) Entity not properly recognized, missing triples in input? http://org.semanticweb.owlapi/error#Error7 for type Class
2018-03-03 17:49:23,435 ERROR (OWLRDFConsumer:1808) Entity not properly recognized, missing triples in input? http://org.semanticweb.owlapi/error#Error8 for type Class
2018-03-03 17:49:23,437 ERROR (OWLRDFConsumer:1808) Entity not properly recognized, missing triples in input? http://org.semanticweb.owlapi/error#Error9 for type Class
2018-03-03 17:49:23,439 ERROR (OWLRDFConsumer:1808) Entity not properly recognized, missing triples in input? http://org.semanticweb.owlapi/error#Error10 for type Class
2018-03-03 17:49:23,441 ERROR (OWLRDFConsumer:1808) Entity not properly recognized, missing triples in input? http://org.semanticweb.owlapi/error#Error11 for type Class
2018-03-03 17:49:23,443 ERROR (OWLRDFConsumer:1808) Entity not properly recognized, missing triples in input? http://org.semanticweb.owlapi/error#Error12 for type Class
2018-03-03 17:49:24,687 ERROR (OWLRDFConsumer:1808) Entity not properly recognized, missing triples in input? http://org.semanticweb.owlapi/error#Error13 for type Class
2018-03-03 17:49:24,689 ERROR (OWLRDFConsumer:1808) Entity not properly recognized, missing triples in input? http://org.semanticweb.owlapi/error#Error14 for type Class
2018-03-03 17:49:24,690 ERROR (OWLRDFConsumer:1808) Entity not properly recognized, missing triples in input? http://org.semanticweb.owlapi/error#Error15 for type Class
2018-03-03 17:49:24,692 ERROR (OWLRDFConsumer:1808) Entity not properly recognized, missing triples in input? http://org.semanticweb.owlapi/error#Error16 for type Class
```

**Comment by graybeal** [comment]
This is a naive answer, but it looks to me (just based on the URIs) like they are not *in* SWEET, they are being generated during the parsing process. The owlapi software is the parsing software, and it's generating a unique error for each issue. But I don't think those are resolvable, so no help in that direction. Also not seeing any line numbers or other useful clues. Might have to do a divide by 2 search, or try another parsing tool to see if the logging is better.

John

**Comment by brandonnodnarb** [comment]
This is related to #122.  I suggest closing this issue as these links were removed from the turtle files (see #123).  

I suggest updating the contributor page to mention this issue, explicitly mentioning the URL http://org.semanticweb.owlapi/error#Error(n), to attempt to rule out one place of origin (new edits).

Perhaps there should be an FAQ and/or checklist for reviewers as well?

**Comment by lewismc** [comment]
https://github.com/ESIPFed/sweet/blob/master/CONTRIBUTING.md#avoiding-presence-of-httporgsemanticwebowlapierrorerrorn-in-contributions-when-editing-using-protege

Thanks @brandonnodnarb

**Comment by cmungall** [comment]
The FAQ states that this is caused by an error in the OWLAPI. Has this been confirmed by a bug report sent to the owlapi tracker?

Note that while I think it's not great behavior for the OWLAPI to inject these Error classes with only warnings on STDERR when it encounters invalid OWL-DL, I think the most likely scenario is still that the sweet import chain contains invalid DL.

I am trying to make a minimal test but this is hard as there are a lot of cyclic imports in sweet. I hypothesize that this is something to do with conflicts between declarations of entity types.

**Comment by cmungall** [comment]
This bug was reported here in 2016: https://github.com/owlcs/owlapi/issues/483

It was fixed in 2016, so it should be in the version of the owlapi that is used by robot. 

However, there are still problems:

```
$ robot convert -i reprMathGraph.ttl -o foo.ttl
2019-11-11 10:04:05,141 ERROR org.semanticweb.owlapi.rdf.rdfxml.parser.OWLRDFConsumer - Entity not properly recognized, missing triples in input? http://org.semanticweb.owlapi/error#Error1 for type Class
2019-11-11 10:04:05,148 ERROR org.semanticweb.owlapi.rdf.rdfxml.parser.OWLRDFConsumer - Entity not properly recognized, missing triples in input? http://org.semanticweb.owlapi/error#Error2 for type Class
2019-11-11 10:04:05,152 ERROR org.semanticweb.owlapi.rdf.rdfxml.parser.OWLRDFConsumer - Entity not properly recognized, missing triples in input? http://org.semanticweb.owlapi/error#Error3 for type Class
2019-11-11 10:04:05,153 ERROR org.semanticweb.owlapi.rdf.rdfxml.parser.OWLRDFConsumer - Entity not properly recognized, missing triples in input? http://org.semanticweb.owlapi/error#Error4 for type Class
2019-11-11 10:04:05,153 ERROR org.semanticweb.owlapi.rdf.rdfxml.parser.OWLRDFConsumer - Entity not properly recognized, missing triples in input? http://org.semanticweb.owlapi/error#Error5 for type Class
```

So either this ticket was closed prematurely, or the one in the owlapi tracker was

**Comment by lewismc** [comment]
I think best thing to do is reopen.

**Comment by dr-shorthair** [comment]
We had a similar issue with trying to load DCAT2 in Protege - it was related to incomplete declaration of some FOAF and DC properties - see

https://github.com/w3c/dxwg/pull/1006
https://github.com/w3c/dxwg/issues/1100

**Comment by ignazio1977** [comment]
> I am trying to make a minimal test but this is hard as there are a lot of cyclic imports in sweet. I hypothesize that this is something to do with conflicts between declarations of entity types.

the bugs I'm aware of about cyclic imports depend on the first ontology opened - i.e., if the first ontology is also part of a cyclic imports closure, that's when the bugs manifest.

Example: 
    A imports B imports C imports B
This does not trigger bugs (not the ones I'm aware of, anyway)
    A imports B imports A
This is where you get troubles.
However, the problem is that declarations have not been parsed fully when axioms in A and B are parsed. Changing the ontology loading to load B instead of A got rid of the problem, in some situations (the loaded imports closure is equivalent, in an import cycle).

This might help to determine which ontology might be patched with extra declarations to work around the problem until the code can be fixed.

**Comment by ignazio1977** [comment]
If it helps, looks like the problems are in these ontologies:

    LoadSweet.main() OntologyID(OntologyIRI(<http://sweetontology.net/reprMath>) VersionIRI(<null>))
    SubClassOf(<http://sweetontology.net/reprMath/OrderedPair> <http://org.semanticweb.owlapi/error#Error1>)

    LoadSweet.main() OntologyID(OntologyIRI(<http://sweetontology.net/reprMathGraph>) VersionIRI(<null>))
    SubClassOf(<http://sweetontology.net/reprMathGraph/Edge> <http://org.semanticweb.owlapi/error#Error10>)
    SubClassOf(<http://sweetontology.net/reprMathGraph/Graph> <http://org.semanticweb.owlapi/error#Error11>)
    SubClassOf(<http://sweetontology.net/reprMathGraph/Graph> <http://org.semanticweb.owlapi/error#Error12>)

    LoadSweet.main() OntologyID(OntologyIRI(<http://sweetontology.net/reprMathOperation>) VersionIRI(<null>))
    SubClassOf(<http://sweetontology.net/reprMathOperation/DefiniteIntegral> <http://org.semanticweb.owlapi/error#Error16>)
    SubClassOf(<http://sweetontology.net/reprMathOperation/IndefiniteIntegral> <http://org.semanticweb.owlapi/error#Error13>)
    SubClassOf(<http://sweetontology.net/reprMathOperation/Integral> <http://org.semanticweb.owlapi/error#Error14>)
    SubClassOf(<http://sweetontology.net/reprMathOperation/Integral> <http://org.semanticweb.owlapi/error#Error15>)

**Comment by lewismc** [comment]
Resolved in #171
